src/net/p2p.rs

```diff
@@ -2896,6 +2927,23 @@ impl PeerNetwork {
                     // go fetch blocks
                     match dns_client_opt {
                         Some(ref mut dns_client) => {
+                            PeerNetwork::with_attachments_downloader(
```

Comment:
This shouldn't be here.  Attachment-downloading should be a parallel state machine, and shouldn't be stealing time from the block-downloader.

Reply:
To see what I mean, look at how the neighbor walk is carried out -- it's a parallel state machine from inv/block synchronization.

Reply:
I'm not sure to understand. The attachments are already being handled in a dedicated state machine, and since all the state machines would be running on the same thread and triggered by the same root invocation (PeerNetwork.run), how could it be stealing time from anyone?

Reply:
The method `do_network_work()` should be _solely_ concerned with essential networking tasks (namely, the tasks enumerated in `NetworkWorkState`).  It should not be used to drive neighbor walking, nor should it be used to drive attachment downloading.  Instead, please encapsulate this functionality into a member method like `do_attachment_downloads()`, and call it from `PeerNetwork::dispatch_network` (like how `do_network_neighbor_walk()` gets called).

The reason for organizing the code this way is that the peer network may need to deprioritize neighbor walks and attatchment downloads in certain conditions.  For example, the peer network doesn't allow the neighbor-walk code to alter the peer graph until it has done a pass through the inventory and block synchronization steps.  As another example, the network code may need to deprioritize attachment synchronization if it detects that it's using too much bandwidth (to the point where it interferes with block downloads).  Keeping the attachment download code as separate as possible from the essential networking tasks makes implementing these priority adjustments much more straight-forward.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2013/commits/8ff08733f8e4a63140037de1cd4bc274651261af